### ISSUE TITLE
bug: @package annotation cannot be use. Closure library throws multiple warnings.

### ISSUE THREAD

**Issue by monzonj** [steida/gulp-closure-compiler#12]
Hello,

Thanks for this great plugin for the closure tools.

There's a bug in the plugin that prevents the use of the @package annotation. If you check closure default implementation, any code can legally invoke a @package protected method if that code is in the same path as the protected method:

https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/jscomp/CodingConventions.java#L305

However, this plugin will use the gulp tempWrite plugin to copy files to random paths breaking this functionality (why this plugin copy them to random paths is beyond my understanding)

I have commented that code (see below) and all works correctly now for me.

```
  var getFlagFilePath = function(files) {
    var src = files.map(function(file) {
      //var relativePath = path.relative(file.cwd, file.path);
      //var tempPath = tempWrite.sync(file.contents.toString(), relativePath);
      return '--js=' + file.path; //tempPath;
    }).join('\n');

    return tempWrite.sync(src);
  };
```

I believe the reason to use `tempWrite` is to avoid super large paths? I'm afraid that breaking closure does not justify the use of `tempWrite`

Regards,

**Comment by steida** [comment]
Thank you, I will take a look.

**Comment by steida** [comment]
tempWrite is for writing into temporally dirs, which are OS specific

**Comment by steida** [comment]
I see, hmm, will fix it asap.

**Comment by steida** [comment]
As I remember, the reason why this plugin is copying closure library files is... historical I guess. There were localization task, which I will have to rewrite again btw, which modified source code directly. I will try to remove it, let see what happens.

**Comment by monzonj** [comment]
I didn't do a pull request because I didn't understand its use. But I just
commented it out because I didn't see the reason to copy the sources to
another path.

If it is any useful, I'm using closure library with  modules (lots of
sources) and advanced compilation.

The plugin works very well (thanks) but I have to run separately
calcdeps.py to get the sources involved from closure library. I might
include that in your pluging, but I'm good for now. :-)

**Comment by steida** [comment]
I already commented it out, it seems it works, except it broked my concatAll gulp-este task. 
As for "calcdeps.py", feel free to send pull request any time :-)
And again, thank you.

**Comment by steida** [comment]
@monzonj Can I see your dev stack somewhere? I am curious about:
1) how to handle thirdparty commonjs compilation warnings
2) is possible to mix goog.provide with CommonJS? I still need a lof of closure library code
3) How it works during development? Do I have to compile?

**Comment by lusever** [comment]
After 0.2.14 this code is not working.
```javascript
gulp.src(jsPaths)
  .pipe(replace('@@INLINE_CSS_IN_JS', cssMin))
  .pipe(closureCompilerThrough)
  .pipe(gulp.dest('./client'))
```
This occurs because the sources are taken from the file and not from the result of gulp-replace.

**Comment by steida** [comment]
Sorry, will fix it asap.

**Comment by steida** [comment]
@lusever Check just published 0.2.15 please.

**Comment by lusever** [comment]
It worked. Thank you!
